arrows/vxl/image_io.cxx

```diff
@@ -138,6 +142,96 @@ convert_image_helper(const vil_image_view<bool>& src,
   dest = src;
 }
 
+
+std::string
+plane_filename(std::string filename, unsigned p)
+{
+  std::string parent_directory =
+    kwiversys::SystemTools::GetParentDirectory(filename);
+  std::string file_name_with_ext =
+    kwiversys::SystemTools::GetFilenameName(filename);
+
+  std::string file_name_no_ext = kwiversys::SystemTools::GetFilenameWithoutLastExtension( file_name_with_ext );
```

Comment:
long line.  please wrap and/or shorten with namespace alias.  We often abbreviate `kwiversys::SystemTools` to `ST`.